data_collecting_plotter:
  ros__parameters:
    NUM_BINS_V: 10
    NUM_BINS_STEER: 10
    NUM_BINS_A: 10
    V_MIN: 0.0
    V_MAX: 11.5
    STEER_MIN: -1.0
    STEER_MAX: 1.0
    A_MIN: -1.0
    A_MAX: 1.0

    wheel_base: 2.79

data_collecting_trajectory_publisher:
  ros__parameters:
    COURSE_NAME: eight_course
    # COURSE_NAME: u_shaped_return
    # COURSE_NAME: straight_line_positive
    # COURSE_NAME: straight_line_negative

    NUM_BINS_V: 10
    NUM_BINS_STEER: 10
    NUM_BINS_A: 10
    V_MIN: 0.0
    V_MAX: 11.5
    STEER_MIN: -1.0
    STEER_MAX: 1.0
    A_MIN: -1.0
    A_MAX: 1.0

    wheel_base: 2.79
    acc_kp: 1.0
    max_lateral_accel: 0.5
    lateral_error_threshold: 2.0
    yaw_error_threshold: 0.50
    velocity_limit_by_tracking_error: 1.0
    mov_ave_window: 100
    target_longitudinal_velocity: 6.0
    longitudinal_velocity_noise_amp: 0.01
    longitudinal_velocity_noise_min_period: 5.0
    longitudinal_velocity_noise_max_period: 20.0

data_collecting_pure_pursuit_trajectory_follower:
  ros__parameters:
    pure_pursuit_type: linearized
    wheel_base: 2.79
    acc_kp: 1.0
    lookahead_time: 2.0
    min_lookahead: 2.0
    linearized_pure_pursuit_steer_kp_param: 2.0
    linearized_pure_pursuit_steer_kd_param: 2.0
    stop_acc: -2.0
    stop_jerk_lim: 2.0
    lon_acc_lim: 2.0
    lon_jerk_lim: 5.0
    steer_lim: 1.0
    steer_rate_lim: 1.0
    acc_noise_amp: 0.01
    acc_noise_min_period: 5.0
    acc_noise_max_period: 20.0
    steer_noise_amp: 0.01
    steer_noise_min_period: 5.0
    steer_noise_max_period: 20.0
